pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/PinotHelixResourceManager.java

```diff
@@ -142,7 +145,8 @@
 
   public PinotHelixResourceManager(@Nonnull String zkURL, @Nonnull String helixClusterName,
       @Nonnull String controllerInstanceId, String dataDir, long externalViewOnlineToOfflineTimeoutMillis,
-      boolean isSingleTenantCluster, boolean isUpdateStateModel, boolean enableBatchMessageMode) {
+      boolean isSingleTenantCluster, boolean isUpdateStateModel, boolean enableBatchMessageMode,
+      String controllerMode) {
```

Comment:
I would recommend separate the Helix Controller out of PinotHelixResourceManager. PinotHelixResourceManager always join as the ADMINISTRATOR. The controllerMode should be handled in the upper layer, and the PinotHelixResourceManager should serve only as Pinot Controller.

Reply:
Extracted Helix controller out of PinotHelixResourceManager. PinotHelixResourceManager join as PARTICIPANT.